task_templates/README.md

```diff
@@ -1,14 +1,55 @@
 # Templates for Custom Tasks
 
-In this directory are three folders - estimators, transforms, and pipelines
+In this directory there are three folders - estimators, transforms, and pipelines. 
+These are collectively known as "tasks". Each of these folders
+contain code snippets, i.e. tasks, that can be uploaded to DataRobot and incorporated into a blueprint. 
+The key "hooks" (see below) are impemented in the main file called custom.py, custom.R, etc. You 
+can use multiple supporting files as well as needed, e.g. to clean up your code with helper functions.  
+Once uploaded each tasks appears as a single box in the Blueprint. 
+The primary difference between these three are their output and their scope:
 
-* estimators - meant to be used as the prediction step in a blueprint
-* transforms - meant to be used as a preprocessing step in a blueprint
-* pipelines - meant to be added as a single task blueprint, these have preprocessing and an
-  estimator built into a single task
+* transforms - transform the input data, e.g. one hot encoding, numeric scaling, etc. 
+  Their output is always a dataframe to be used by other transforms or estimators.
```

Comment:
we could mention here that output also includes headers/column names (it's a common question and many users don't realize that)